apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClientBuilder.cs

```diff
@@ -0,0 +1,153 @@
+
+    /// <summary>
+    /// Specifies how to respond to the presence of emulator environment variables.
+    /// </summary>
+    /// <remarks>
+    /// This property defaults to <see cref="EmulatorDetection.None"/>, meaning that
+    /// environment variables are ignored.
+    /// </remarks>
+    public new EmulatorDetection EmulatorDetection
+    {
+        get => base.EmulatorDetection;
+        set => base.EmulatorDetection = value;
+    }
+
+    /// <summary>
+    /// Creates a new instance with default settings. <see cref="TopicName"/> must be set
+    /// before calling either <see cref="Build"/> or <see cref="BuildAsync"/>.
+    /// </summary>
+    public PublisherClientBuilder() : base(PublisherServiceApiClient.ServiceMetadata)
+    {
+    }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <returns></returns>
+    /// <exception cref="NotImplementedException"></exception>
+    public override PublisherClient Build() =>
+    // With isAsync set to false, the returned task will already be completed(either successfully or faulted),
+    // so .ResultWithUnwrappedExceptions() will always return immediately.
+        BuildAsyncImpl(default, false).ResultWithUnwrappedExceptions();
+
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    /// <exception cref="NotImplementedException"></exception>
+    public override Task<PublisherClient> BuildAsync(CancellationToken cancellationToken = default) =>
+        BuildAsyncImpl(cancellationToken, true);
+
+    private async Task<PublisherClient> BuildAsyncImpl(CancellationToken cancellationToken, bool isAsync)
+    {
+        Validate();
+        var clientCount = ClientCount ?? Environment.ProcessorCount;
+        var clients = new PublisherServiceApiClient[clientCount];
+        var settings = Settings?.Clone() ?? new PublisherClient.Settings();
+        var shutdowns = new Func<Task>[clientCount];
+        for (int i = 0; i < clientCount; i++)
+        {
+            // Use a random arg to prevent sub-channel re-use in gRPC, so each channel uses its own connection.
+            var grpcChannelOptions = s_unlimitedSendReceiveChannelOptions
+                .WithCustomOption("sub-channel-separator", Guid.NewGuid().ToString());
+
+            var builder = new PublisherServiceApiClientBuilder(this, grpcChannelOptions);
+            clients[i] = isAsync
+                ? await builder.BuildAsync(cancellationToken).ConfigureAwait(false)
+                : builder.Build();
+            var channel = builder.LastCreatedChannel;
+            shutdowns[i] = channel is null ? () => Task.CompletedTask : channel.ShutdownAsync;
+        }
+        Func<Task> shutdown = () => Task.WhenAll(shutdowns.Select(x => x()));
+        return new PublisherClientImpl(TopicName, clients, settings, shutdown);
+    }
+
+    /// <inheritdoc />
+    protected override ChannelPool GetChannelPool() => throw new NotImplementedException();
```

Comment:
Can you override CanUseChannelPool to always return false as well?

Reply:
There's no need - we'll never end up calling that, because we'll never call CreateChannel on this class.